src/Files.App/Filesystem/RecentItems.cs

```diff
@@ -187,19 +189,19 @@ public bool ClearRecentItems()
 		/// This will also unpin the item from the Recent Files in File Explorer.
 		/// </summary>
 		/// <returns>Whether the action was successfully handled or not</returns>
-		public bool UnpinFromRecentFiles(string path)
+		public Task<bool> UnpinFromRecentFiles(RecentItem item)
 		{
-			try
-			{
-				var command = $"-command \"((New-Object -ComObject Shell.Application).Namespace('shell:{QuickAccessGuid}\').Items() " +
-							  $"| Where-Object {{ $_.Path -eq '{path}' }}).InvokeVerb('remove')\"";
-				return Win32API.RunPowershellCommand(command, false);
-			}
-			catch (Exception ex)
+			return SafetyExtensions.IgnoreExceptions(() => Task.Run(async () =>
 			{
-				App.Logger.Warn(ex, ex.Message);
+				using var pidl = new Shell32.PIDL(item.PIDL);
```

Comment:
Improvement: remove item directly instead of calling powershell. Wrap in Task.Run to avoid blocking UI.